src/sspilib/_sec_context.py

```diff
@@ -329,29 +333,24 @@ class ClientSecurityContext(SecurityContext):
     specified. Custom flags can be requested but it will overwrite the default
     set.
 
-    The credential stores the credential to use for authentication as well as
-    extra information that controls how the security protocol will perform
-    the authentication steps.
-
     Optional channel bindings can be specified which is used to tie the
     security context with an outer channel.
 
     Properties like ``expiry`` and ``attributes`` won't be populated until the
     context has been negotiated and is complete.
 
     Args:
+        credential: A credential to use for authentication.
         target_name: The target service's name, typically the SPN.
         flags: Custom ISC REQ flags to use.
-        credential: A credential to use for authentication, defaults to the
-            current user context if not set.
         channel_bindings: Optional channel bindings to tie the context to.
     """
 
     def __init__(
         self,
+        credential: raw.CredHandle,
```

Comment:
Sorry, I did not catch this earlier, but placing `credential` after `target_name` is probably a less disruptive change. `target_name` was a single required argument, so in my code I had `ClientSecurityContext(spn, credential=UserCredential())`, which will be broken by this change. I now updated to `ClientSecurityContext(target_name=spn, credential=UserCredential())`, but making credential the second argument would work without any changes.

Reply:
Yea I did consider this and found that having it first aligns this with the server context object and as it was already a breaking change and I couldn't find any active examples that used the failing way in GitHub search I decided to just go with it.

Reply:
I found one example in pyinstaller's tests: https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/5a553ff210635443d41ed5dab97f47196fcfa8e3/src/_pyinstaller_hooks_contrib/tests/test_libraries.py#L1770

Reply:
Which is actually the only use I've found outside of this repo :)

Reply:
Thanks for finding that, I'll submit a PR to use the syntax that will work with both versions. I prefer to keep it consistent especially in the early stages of this project and in IMO people would probably expect the credentials to be the first thing they would supply.

Reply:
https://github.com/pyinstaller/pyinstaller-hooks-contrib/pull/712